File: tests/files/extract_func_works_before.rb
class F
  def f
    return 'f'
  end
end

def foo(a)
  for j in 1..5 do
    b = 'b'
    c = 'c'
    d = ['d', b]
    e = {'e' => 'e'}
    f = F.new
    g, h = ->() {"g"}, "h"
    k = 'k'
    l = 'l'
    m = 'm'

    a = a
    a = a + 1
    a+=a
    a++
    print b
    print c + 1
    print d[0]
    print e['e']
    print f.f()
    print g.call()
    if h then end
    while k do end
    until l do end
    loop do break if m end
    [a,b].each do |v| puts "#{a} #{v}" end
    print(j)

    return a
  end
end

File: tests/files/extract_func_works_before.java
class F {
    public String f() {
        return "f";
    }
}

record E (String e) {}

class Foo {
    public String foo(int a) {
        String i;
        for (int j = 0; j < 5; j++) {
            String b = "b";
            String c = "c";
            String[] d = {"d"};
            E e = new E("e");
            F f = new F();
            boolean g = true, h = false;
            boolean k = true;
            boolean l = true;

            a = a + 1;
            a += a;
            a++;
            ++a;
            System.out.println(b);
            System.out.println(c + 1);
            System.out.println(d[0]);
            System.out.println(e.e());
            System.out.println(f.f());
            System.out.println(g);
            if (h) {}
            while (k) {}
            do {} while (l);
            i = "i";
            System.out.println(j);

            return i;
        }
        return "";
    }
}
